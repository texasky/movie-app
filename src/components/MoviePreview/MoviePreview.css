.movie-preview {
    display: flex;
    margin-top: 130px;
    position: relative;
    justify-content: center;
    padding-left: 60px;
}

.movie-preview .back-button {
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
}

.movie-preview-card {
    display: flex;
    background-color: rgba(100,149,237, 0.7);
    border: 3px solid royalblue;
    border-radius: 5px;
    margin: 0 50px 0 50px;
    text-align: left;
}

.movie-preview-card img {
    border-radius: 5px 0 0 5px;
}

.movie-preview-card span {
    margin: 10px 0;
}

.movie-preview-info {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    padding: 15px;
}

@media screen and (max-width: 700px) {
    .movie-preview-card {
        flex-direction: column;
        margin-left: 40px;
    }

    .movie-preview .back-button {
        margin-left: 10px;
    }
}