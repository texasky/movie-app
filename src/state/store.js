import { applyMiddleware, createStore } from "redux"
import configureStore from 'redux-mock-store'
import reducers from "./reducers/index"
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

const store = createStore(reducers, composeWithDevTools(applyMiddleware(thunk)));

export default store; 