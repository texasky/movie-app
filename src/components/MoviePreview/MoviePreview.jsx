import { useSelector } from 'react-redux';
import BackButton from '../BackButton/BackButton';
import './MoviePreview.css'

const MoviePreview = (props) => {
    const moviesData = useSelector(state => state.moviesStore.movies);
    const openedMovieId = useSelector(state => state.moviesStore.openedMovieId);
    const openedMovieData = moviesData.filter(mov => mov.imdbID === openedMovieId)[0];

    return (
        <div className='movie-preview'>
            <BackButton/>
            <div className='movie-preview-card'>
                <img src={openedMovieData.Poster} alt={openedMovieData.Title}/>
                <div className='movie-preview-info'>
                    <span>Title: {openedMovieData.Title}</span>
                    <span>Year: {openedMovieData.Year}</span>
                    <span>Type: {openedMovieData.Type}</span>
                </div>
            </div>
        </div>
    )
}

export default MoviePreview;