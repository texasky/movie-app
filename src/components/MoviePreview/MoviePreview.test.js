import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MoviePreview from './MoviePreview';
import { Provider } from 'react-redux';
import store from './../../state/store'

test('Movi preview component tests', () => {
    const utils = render(<Provider store={store}><MoviePreview /></Provider>);

    const movieHeader = utils.getByTestId('movie-preview');

    expect(movieHeader).toBeInTheDocument();
});

test('Movi preview component matches snapshot', () => {
    const tree = renderer.create(<Provider store={store}><MoviePreview /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});