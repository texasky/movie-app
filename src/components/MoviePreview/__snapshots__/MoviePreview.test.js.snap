// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movi preview component matches snapshot 1`] = `
<div
  className="movie-preview"
  data-testid="movie-preview"
>
  <div
    className="back-button"
    data-testid="back-button"
    onClick={[Function]}
  />
  <div
    className="movie-preview-card"
  >
    <img />
    <div
      className="movie-preview-info"
    >
      <span>
        Title: 
      </span>
      <span>
        Year: 
      </span>
      <span>
        Type: 
      </span>
    </div>
  </div>
</div>
`;
